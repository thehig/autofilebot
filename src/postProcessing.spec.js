console.log(""); // use log before requiring mockfs to prevent 'callsites' error
const mock = require("mock-fs");
const fs = require("fs");

const config = require("config");
const tempDir = config.get("temp"); //?
const toDir = config.get("to"); //?
const logfile = config.get("log"); //?

const fileStructure = {
  [tempDir]: {
    "Ash vs Evil Dead - 3x06 - Tales from the Rift.mkv": ""
  },
  [toDir]: {
    "Ash vs Evil Dead": {
      "Ash vs Evil Dead - 3x01 - Family.mkv": "",
      "Ash vs Evil Dead - 3x02 - Booth Three.mkv": "",
      "Ash vs Evil Dead - 3x03 - Apparently Dead.mkv": "",
      "Ash vs Evil Dead - 3x04 - Unfinished Business.mkv": "",
      "Ash vs Evil Dead - 3x05 - Baby Proof.mkv": ""
    }
  }
};

const { walk } = require("./fileOperations");
const postProcess = require("./postProcessing");

const nyi = done => done.fail("Not Yet Implemented");

describe.only("postProcessing", () => {
  beforeEach(() => {
    mock(fileStructure);
  });

  afterEach(() => {
    mock.restore();
  });

  describe("errors if", () => {
    it("missing fromDir", done => {
      postProcess()
        .then(() => done.fail())
        .catch(err => {
          expect(err.message).toContain("missing");
          done();
        });
    });

    it("missing toDir", done => {
      postProcess("someparam")
        .then(() => done.fail())
        .catch(err => {
          expect(err.message).toContain("missing");
          done();
        });
    });
  });

  it("moves file from temp to destination dir", () => {
    const before = () =>
      walk(tempDir)
        .then(f => expect(f).toHaveLength(1))
        .then(() => walk(toDir))
        .then(f => expect(f).toHaveLength(5));

    const after = () =>
      walk(tempDir)
        .then(f => expect(f).toHaveLength(0))
        .then(() => walk(toDir))
        .then(f => expect(f).toHaveLength(6));

    return before()
      .then(() => postProcess(tempDir, toDir))
      .then(() => after());
  });
});
