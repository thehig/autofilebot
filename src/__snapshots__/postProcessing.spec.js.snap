// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postProcessing moves multiple files from temp to existing and non-existing destinations 1`] = `
Array [
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Archer (2009)\\\\Archer (2009) - 8x02 - Archer Dreamland Berenice.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Archer (2009)\\\\Archer (2009) - 8x03 - Jane Doe.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Dr Who\\\\Doctor Who (2005) - 10x02 - Smile.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Dr Who\\\\Doctor Who (2005) - 8x04 - Listen.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Last Week Tonight\\\\Last Week Tonight with John Oliver - 1x11 - Episode 11.mp4",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Last Week Tonight\\\\Last Week Tonight with John Oliver - 1x16 - Episode 16.mp4",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Marvels Agents of Shield\\\\Marvel's Agents of S.H.I.E.L.D. - 2x01 - Shadows.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Marvels Agents of Shield\\\\Marvel's Agents of S.H.I.E.L.D. - 5x21 - The Force of Gravity.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Planet Earth 2\\\\Planet Earth II - 1x02 - Mountains.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\Planet Earth 2\\\\Planet Earth II - 1x03 - Jungles.mkv",
  "C:\\\\autofilebot\\\\fixtures\\\\to\\\\The Marvelous Mrs. Maisel\\\\The Marvelous Mrs. Maisel - 1x05 - Doink.mkv",
]
`;
